### DISCUSSION TITLE
Is it possible to create a calculated measure in Pygwalker?

### DISCUSSION THREAD

**Discussion by ddealwis09** [Kanaries/pygwalker#207] [General]
Wonder if measures can be created from the base values, for example market share? 

thanks!

**Comment by longxiaofei** [comment]
We plan add customize measure in 0.4.x version.

You can use sql to flexibly set your measure.

Do you have any suggestions?

**Reply by 353913575** [reply]
Can the calculation field, which first aggregates two columns based on the selected dimension in visualization and then divides them, be implemented in version 0.4.x?

**Reply by longxiaofei** [reply]
@353913575  Hi, Could you give an example?

**Reply by 353913575** [reply]
For example, if I want to calculate the efficiency of equipment, this metric would be defined as eff = sum(re_time)/sum(ty_time). This metric requires aggregation before division. I would like to create such a calculated field in Pygwalker so that when I drag different dimensions, the field will aggregate and then divide according to the dimensions I have dragged,I look forward to your reply,thanks

**Reply by longxiaofei** [reply]
Yes, This is exactly what a calculated measure does.

**Reply by 353913575** [reply]
This is exhilarating news. I'm already looking forward to using this feature, thank you, super developer!!!

**Reply by fabienval** [reply]
Would we also be able to "chain" aggregation?
My example would be a dataframe of itemised sales and what I would want to be able to display is:
The average (hourly sum of sales) for each day of the week, while still being able to filter for specific products.
Thanks

**Reply by longxiaofei** [reply]
![image](https://github.com/Kanaries/pygwalker/assets/28337703/5f21370d-799b-459e-9940-6a4bb211dcb7)

Like this? Currently it can be implemented without calculating fields, or maybe I misunderstood what you mean.

@fabienval
